### PR TITLE
experimental sleep for power saving

### PR DESCRIPTION
**Pull request by zenrandom**
yeah, i have no idea if this will work or not... but maybe :)

### CODE REVIEW

**Comment by devsecfranklin** [comment]
Trying to figure out why it's saying this: 

/home/thedevilsvoice/workspace/bat_mini/arduino/bat_mini_2018/bat_mini_2018.ino: In function 'void system_sleep()':
bat_mini_2018:62: error: 'cbi' was not declared in this scope
   cbi(ADCSRA,ADEN);                    // switch Analog to Digitalconverter OFF
                  ^
bat_mini_2018:70: error: 'sbi' was not declared in this scope
   sbi(ADCSRA,ADEN);                    // switch Analog to Digitalconverter ON

**Comment by zenrandom** [comment]
small change, removed cbi/sbi for adc enable/disable... using different syntax

**Review by zenrandom — COMMENTED** [review]
under the idea that sleep is non blocking the sleep time has been changed to 64ms from 128ms and the delay 100 put back, the processor should be sleeping 60% of the delay...

File: arduino/bat_mini_2018/bat_mini_2018.ino [hunk 1]
```diff
@@ -129,6 +189,7 @@ void flicker() {
 void sin_wave() {
 
   mybat.buttonState = HIGH;
+  system_sleep();
```
Comment (devsecfranklin) [inline arduino/bat_mini_2018/bat_mini_2018.ino:184]: should this be after the delay? or does it matter?

File: arduino/bat_mini_2018/bat_mini_2018.ino [hunk 2]
```diff
@@ -148,18 +209,21 @@ void sin_wave() {
       ibright = int(sin(tcount) * 255);
       colors[i] = mybat.hsvToRgb(ihue, 255, 255);
       ledStrip.write(colors, LED_COUNT, ibright);
-      delay(100);
+      system_sleep();
```
Comment (devsecfranklin) [inline arduino/bat_mini_2018/bat_mini_2018.ino:204]: should this be after the delay? or does it matter?

File: arduino/bat_mini_2018/bat_mini_2018.ino [hunk 3]
```diff
@@ -148,18 +209,21 @@ void sin_wave() {
...
+	  delay(100);
     }
 
     ledStrip.write(colors, LED_COUNT, 10);
-    delay(100);
+    system_sleep();
```
Comment (devsecfranklin) [inline arduino/bat_mini_2018/bat_mini_2018.ino:209]: should this be after the delay? or does it matter?

**Review by devsecfranklin — COMMENTED** [review]
still doing the thing where the heart flashes super fast. 

Going to try testing again a different way.

**Comment by devsecfranklin** [comment]
So with the recent changes I pushed:
- the button works again
- the heartbeat is not the correct tempo (works but timing is off) 
- The 4th pattern  rainbow() the heart goes solid and the eyes do not blink. 
- The tri color LEDs work but the timing seems off a bit. (Like the transitions are less smooth? 

Perhaps we could take the sleep timer out of rainbow() since that needs the smooth light gradients?

**Pull request closed without merging**